### ISSUE TITLE
Getting too many cache misses and high cache put after a period 

### ISSUE THREAD

**Issue by jpbassinello** [redisson/redisson#2899]
Our solution uses Redisson configuration with a programmatic cache creation using `org.redisson.jcache.JCacheManager` with a single Redis server.
Caches are created with `cache-api` configuration and a ExpiryPolicy.

```
private Configuration<Object, Object> cacheConfig(final Factory<ExpiryPolicy> expiryPolicyFactory) {
		var configuration = new MutableConfiguration<>()
				.setExpiryPolicyFactory(expiryPolicyFactory)
				.setStatisticsEnabled(true);

		return RedissonConfiguration.fromInstance(redissonClient, configuration);
}
```

When the application loads for the first time, caches started to be used and for a period they work as expected with high cache hits.

After a period of tests, one (or many) caches start "deteriorating" and cache miss ratio starts to be higher. Cache put time also started to be very high.
We have a screen showing cache statistics. This is one example of statistics after the problem occurs.

![Screenshot_2](https://user-images.githubusercontent.com/3186105/86831718-a034bb00-c06d-11ea-9d97-c439eb7899b9.png)
At the example, all other caches of same type are working as expected. Just "Usuario.funcionalidades" is having the problem.

The problem always happens with a nested collection in a Hibernate entity.
![Sem título](https://user-images.githubusercontent.com/3186105/86831978-f6a1f980-c06d-11ea-883d-241f8f96bd95.png)

Have you seen similar behavior? Any clues about how to understand better what is happening?

**Comment by mrniko** [comment]
> Cache put time also started to be very high.

try to run slowlog to realize the cause of it

**Comment by jpbassinello** [comment]
The problem was solved upgrading from `3.12.0` to `3.13.2`.
